api/src/main/java/org/openmrs/CodedOrFreeText.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * A Concept object can represent either a question or an answer to a data point. That data point is
+ * usually an {@link Obs}. <br>
+ * <br>
+ * ConceptName is the real world term used to express a Concept within the idiom of a particular
+ * locale.
+ * <br>
+ * The purpose of this class therefore is to record information of a single encounter by taking the
+ * concept and concept name or a string for a concept that may not be in the database.
+ */
+
+public class CodedOrFreeText {
+	
+	Concept coded;
+	
+	ConceptName specificName;
+	
+	String nonCoded;
+	
+	/** Default constructor */
+	public CodedOrFreeText() {
+	}
+	
+	/**
+	 * Convenience constructor with concept, concept name and manually written concept name to save
+	 *
+	 * @param coded the concept object to set
+	 * @param specificName the specificName of the concept to set
+	 * @param nonCoded the manually written string name of the concept to set
+	 */
+	public CodedOrFreeText(Concept coded, ConceptName specificName, String nonCoded) {
+		this.coded = coded;
+		this.specificName = specificName;
+		this.nonCoded = nonCoded;
+	}
+	
+	/**
+	 * Return the Concept object
```

Comment:
Gets the coded concept, would be better.

Reply:
Noted